### PR TITLE
REF/TST: glm links

### PR DESCRIPTION
**Pull request by josef-pkt**
first commit here is to silence the unit test warnings.
code and tests are adjusted to avoid deprecated link names
back to 3 warnings when running all genmod unit tests
1 negbin warning still to silence

includes option to turn of link check,  closes #1880 (previous PR)

### CODE REVIEW

**Comment by josef-pkt** [comment]
discrete tests have 70 warnings, but none are future warnings for links (I expected at least some)
most warnings are L1 fit_regularized, plus several or many convergence warnings.

there should be GLM based tests also in other directories.

**Comment by josef-pkt** [comment]
I tried to figure out some warnings in discrete unit tests. Too time consuming.
for example
I add catch_warnings in test_truncated_model.py fit_regularized but it does not catch all ConvergenceWarnings. I cannot figure out why not.

merging when green

File: statsmodels/discrete/tests/test_truncated_model.py [hunk 1]
```diff
@@ -45,9 +51,12 @@ def test_bic(self):
 
     def test_fit_regularized(self):
         model = self.res1.model
-
         alpha = np.ones(len(self.res1.params))
-        res_reg = model.fit_regularized(alpha=alpha*0.01, disp=0)
+
+        with warnings.catch_warnings():
```
Comment (bashtage) [inline statsmodels/discrete/tests/test_truncated_model.py:56]: Any reason not to use `with pytest.warns(UserWarning)` to simplify?

Comment (josef-pkt) [reply statsmodels/discrete/tests/test_truncated_model.py:56]: It shouldn't raise a warning. This is just to reduce a bit of test noise.
fit_regularized failures should be investigated and cleared up, but I never took the time for it for the new count models.

**Comment by josef-pkt** [comment]
imported ConvergenceWarning twice, amended and force pushed

**Comment by josef-pkt** [comment]
all green, unrelated failures (the usual)

**Pull request merged**